src/components/solvers/SolverPicker.tsx

```diff
@@ -1,49 +1,83 @@
-import React, {useEffect, useState} from "react";
-import {Select, Text, Tooltip} from "@chakra-ui/react";
-import {fetchSolvers} from "../../api/ToolboxAPI";
-import {ProblemSolver} from "./ProblemSolver";
+import {Box, Container, HStack, Select, Text, Tooltip} from "@chakra-ui/react";
+import React, { ChangeEvent, useEffect, useState } from "react";
+import { fetchSolvers, fetchSubRoutines } from "../../api/ToolboxAPI";
+import { SubRoutineDefinition } from "./SubRoutineDefinition";
+import { ProblemSolver } from "./ProblemSolver";
+import { SolveRequest } from "./SolveRequest";
 
 export interface SolverPickerProps {
-    problemType: string;
-    setSolver?: (solver: ProblemSolver | undefined) => void;
+    problemUrl: string;
+    problemDescription?: string;
+    setSolveRequest: (subRoutines: SolveRequest) => void;
 }
 
 export const SolverPicker = (props: SolverPickerProps) => {
     const [loadingSolvers, setLoadingSolvers] = useState<boolean>(true);
     const [solvers, setSolvers] = useState<ProblemSolver[]>([]);
+    const [subRoutines, setSubRoutines] = useState<SubRoutineDefinition[] | undefined>(undefined);
+    const [solveRequest, setSolveRequest] = useState<SolveRequest>({
+        requestedSubSolveRequests: {}
+    });
 
     useEffect(() => {
+        setSubRoutines(undefined);
         setLoadingSolvers(true);
-        fetchSolvers(props.problemType)
-            .then(solvers => {
+        fetchSolvers(props.problemUrl)
+            .then((solvers: ProblemSolver[]) => {
                 setSolvers(solvers);
                 setLoadingSolvers(false);
             })
-    }, [props.problemType]);
+    }, [props.problemUrl]);
+
+    function onSolverChanged(e: ChangeEvent<HTMLSelectElement>) {
+        if (e.target.selectedIndex == 0 || e.target.selectedIndex > solvers.length) {
+            setSubRoutines(undefined);
+            props.setSolveRequest?.(solveRequest)
+        } else {
+            let solver = solvers[e.target.selectedIndex - 1];
+            solveRequest.requestedSolverId = solver.id;
+            fetchSubRoutines(props.problemUrl, solver.id)
+                .then(subRoutines => setSubRoutines(subRoutines));
+            props.setSolveRequest?.(solveRequest)
+        }
+    }
 
     const getSolvers = () => {
```

Comment:
This should have a react component name (for example `SolverSelector`)

Reply:
You mean it shouldn't be `SolverPicker`, because `Picker` isn't usually used for naming components?

Reply:
Well it shouldn't be `SolverPicker` because the whole component already has that name (and `SolverPicker` is a good name IMO). I just wanted to say that `getSolvers` doesn't sound like and isn't spelled like a React Component name.

Reply:
I see. I was honestly a bit confused because I didn't initially realize you mean that `getSolvers` is a component.
Having them nested like that, I just treated it like a function, hence the name `getSolvers`.

Is there a preferred way of doing this?
`function getSolverSelection() { ... }`
vs
`const SolverSelection = () => { ... }`

Reply:
Oh, that's probably my bad.

`SolverSelection` would be more common as you can use these components just like any other:

`<SolverSelection/>` instead of `getSolvers()` (line 66)

Reply:
Ok, sounds good!